modules/core/include/opencv2/core/types.hpp

```diff
@@ -2432,6 +2480,111 @@ inline
 TermCriteria::TermCriteria(int _type, int _maxCount, double _epsilon)
     : type(_type), maxCount(_maxCount), epsilon(_epsilon) {}
 
+inline TermCriteria::TermCriteria(int count, double _epsilon)
+    : type(TermCriteria::MAX_ITER), maxCount(count), epsilon(_epsilon)
+{
+    if (maxCount < 0)
+    {
+        maxCount = std::numeric_limits<int>::max();
+    }
+
+    if (epsilon > 0)
+    {
+        type += TermCriteria::EPS;
+    }
+}
+
+inline bool TermCriteria::fromDeprecated()
+{
+    if (!isValid())
+    {
+        return false;
+    }
+    else
+    {
+        const bool isMaxCount = type & COUNT;
+        const bool isEps = type & EPS;
+
+        if (!isMaxCount)
+        {
+            maxCount = std::numeric_limits<int>::max();
+        }
+
+        if (!isEps)
+        {
+            epsilon = -1;
+        }
+
+        return true;
+    }
+}
+
+inline bool TermCriteria::checkEpsilonTolerance(double _epsilon) const
+{
+    if ((epsilon >= 0) && (_epsilon > epsilon))
```

Comment:
Criteria can be simplified.
`return (epsilon <= 0.) | (_epsilon < epsilon)`.

`operator | ` is leaved intentionally, because it helps to avoid branch in short-circuit evaluations.